caikit/runtime/http_server.py

```diff
@@ -178,12 +181,8 @@ def shutdown_wrapper(*args, **kwargs):
         # Placeholder for thread when running without blocking
         self._uvicorn_server_thread = None
 
-    def __del__(self):
```

Comment:
Did this move somewhere?

Reply:
So, having this around was giving an issue after a successful test run in which the `flush_metrics` was trying to write to a file that was already closed. 
```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/logging/__init__.py", line 1082, in emit
    stream.write(msg + self.terminator)
ValueError: I/O operation on closed file.
```

I figured if we want to keep this, we should have a `flag` in `global_predict_servicer.stop_metering()` that checks if we've already stopped metering before trying to `flush_metrics`?